### DISCUSSION TITLE
Esp32 Connection

### DISCUSSION THREAD

**Discussion by mexusbg** [S-CODE-pl/ChainBreaker#4] [General]
I would like to make a DIY BT adapter with ESP32 so I can make adjustments on the fly. I can make it open source but will need an app for it. Can you contact me so we can discuss in details if interested?

**Comment by Tomblarom** [comment]
I thought about doing the same! Are you interested to work together? Maybe we can combine it with the PC software: https://github.com/andrey-pr/OpenBafangTool He currently works on CAN motors and maybe we could build the ESP32-C3 or C6 firmware to introduce the gateway, for programming. I hate that Bafang requires the BESST, with limited accounts and workarounds.. btw we have a telegram group: https://t.me/bafang_m500_m600_development

Instead of a dongle, we could add a display and make a OpenSource Bafang replacement display :)

**Reply by mexusbg** [reply]
I have done some projects with serial over WiFi or BLE. I need to dig them out of my old desktop. I can test the hardware (must find the cables, I've misplaced years ago) and design a 3d printed case.  I just need to have the app the option to select the BLE or Wifi device for communication. The rest is easy.

We use two serial connections, sending data from one to the other, and we can send info to either one wirelessly.

About the display it won't be to hard, but I don't have so much free time. And designing and machining strong mounts will be a hard task,

**Reply by Tomblarom** [reply]
I'm very good in designing for 3D-printing in CAD. I'm thinking about something like the [Bafang DPC245](https://bafang-e.com/de/oem-area/components/component/hmi/dp-c245can) and would work with ESP32C3 + TJA1050, since its very easily available. I did a similar project for VESC: https://forum.esk8.news/t/usb-c-to-vesc-can-forward-adapter-cheap-and-diy/79789/29

**Reply by Tomblarom** [reply]
@mexusbg are you still interested? There is another guy that works with me. We do an OpenSource project CAN WiFi/BT bridge to connect via OpenBafangTool. Join us! https://t.me/+rca7AzMFTZw1MTAy

**Reply by mexusbg** [reply]
> @mexusbg are you still interested? There is another guy that works with me. We do an OpenSource project CAN WiFi/BT bridge to connect via OpenBafangTool. Join us! https://t.me/+rca7AzMFTZw1MTAy

Telegram me at: @mexusbg
